spec/services/versioning/revision_updater/image_spec.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+RSpec.describe Versioning::RevisionUpdater::Image do
+  let(:user) { create :user }
+  let(:image_revision) { create :image_revision }
+
+  describe "#update_image" do
+    it "extends image revisions with a new image" do
+      revision = create(:revision, image_revisions: [image_revision])
+      new_image = create :image_revision
+
+      updater = Versioning::RevisionUpdater.new(revision, user)
+      updater.update_image(new_image)
+
+      next_revision = updater.next_revision
+      expect(next_revision.image_revisions).to match_array [image_revision, new_image]
+    end
+
+    it "updates an existing image with a new revision" do
+      revision = create(:revision, image_revisions: [image_revision])
+      updated_image = create :image_revision, image_id: image_revision.image_id
```

Comment:
Nitpick, but bit jarring to have the two different parenthesis styles here.

Reply:
There seem to be a few of these in this file.

Reply:
Oh yeah, I've gone without parentheses as they're not necessary. I remember we thought about getting the linter to check for this :-|.